sys/mmap.c

```diff
@@ -38,7 +38,7 @@ vm_addr_t do_mmap(vm_addr_t addr, size_t length, vm_prot_t prot, int flags,
 
   vm_map_entry_t *entry;
 
-  WITH_RW_LOCK (&vmap->rwlock, RW_WRITER) {
+  WITH_MTX_LOCK (&vmap->mtx) {
```

Comment:
Could you devise more specialized tool like: `WITH_VMMAP_LOCKED(vmap)`?
Let's not use directly exposed fields of `vm_map_t` structure.